### PR TITLE
Jenkinsfile for configuring pipeline

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

File: .jenkins [hunk 1]
```diff
@@ -0,0 +1,52 @@
+pipeline {
+    agent {
+        label 'DoNotRunAnythingOnMaster'
+    }
+    stages {
+        stage('Build') {
+            parallel {
+                stage('CUDA-10.1-NVCC') {
```
Comment (aprokop) [inline .jenkins]: Why call it `NVCC` if it's already `CUDA`?

File: .jenkins [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                    steps {
+                        sh 'rm -rf build && mkdir -p build && cd build && cmake -D CMAKE_BUILD_TYPE=Debug -D CMAKE_CXX_COMPILER=$KOKKOS_DIR/bin/nvcc_wrapper -D CMAKE_PREFIX_PATH="$KOKKOS_DIR;$BOOST_DIR;$BENCHMARK_DIR" -D MPIEXEC_PREFLAGS="--allow-run-as-root" .. && make -j8 VERBOSE=1 && ctest --output-on-failure'
+                    }
+                }
+                stage('Clang') {
+                    agent {
+                        dockerfile {
+                            filename "Dockerfile"
+                            dir "docker"
+                            additionalBuildArgs '--build-arg BASE=ubuntu:18.04 --build-arg KOKKOS_OPTIONS="--with-serial --with-openmp --arch=AMD --compiler=clang++"'
```
Comment (aprokop) [inline .jenkins]: Why `AMD` arch?

**Comment by aprokop** [comment]
@dalg24 You want to clean the history of this PR?

File: .jenkins (hunk 1 above)
Comment (dalg24) [reply .jenkins]: cause I planned on having clang + cuda

File: .jenkins (hunk 2 above)
Comment (dalg24) [reply .jenkins]: It is what the machine (cathode) is

**Comment by masterleinad** [comment]
So building with `clang` worked out of the box for you? Must really be an issue with boost-1.70.0 then.

**Comment by dalg24** [comment]
> So building with `clang` worked out of the box for you? Must really be an issue with boost-1.70.0 then.

Clang without CUDA at the moment.  (We are running out of cycles on the slave machines with GPUs...)

**Comment by masterleinad** [comment]
I wonder if `CUDA` really is the problem for me. I'd like to figure that out later today.

**Pull request merged**